### PR TITLE
Add types to AlertService

### PR DESCRIPTION
**Pull request by gmarziou**
Just playing with types.

### CODE REVIEW

File: src/service/alert.service.ts [hunk 1]
```diff
@@ -2,11 +2,37 @@ import { Injectable, Sanitizer, SecurityContext } from '@angular/core';
...
+    msg: string;
+    params?: any;
+    timeout: number;
+    toast: boolean;
+    position: string;
+    scoped: boolean;
+    close: (alerts: Alert[]) => void;
+}
+
+export class AlertOptions {
```
Comment (deepu105) [inline src/service/alert.service.ts]: This class can extend Alert and add just alertID I guess or all common items can be in BaseAlert and both Alert and AlertOptions can extend that. I also need to check if alertId is really required may be we can just use one Alert class

Comment (gmarziou) [inline src/service/alert.service.ts]: Yes what I found is that you add types to existing JS code, it's always makes you re-think this code.
I wanted to keep them as 2 classes as it seemed that some some fileds of AlertOptions were optional when they were not in Alert.

**Comment by gmarziou** [comment]
Also I noticed that you kept the presentation part of the alerts in the generated code probably to let user define how they look but it seems this code could be a good candidate for being a self contained component, maybe offering only CSS styling.

**Comment by deepu105** [comment]
Yes it could be, But didnt have enough time to migrate it. Also there was some stuff it was using from interceptors and things which the user might wanna change. but we could see if we can move it here and still let users configure it if required

**Pull request merged**